### PR TITLE
Added prettier, husky, lint-staged

### PR DESCRIPTION
**Pull request by jamonholmgren**
I've added prettier and run through the whole code base with it. Also added husky to run `lint-staged` before committing new files. I'll be adding CI linting too, once we get this merged in.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -88,8 +85,17 @@
...
-    "node": ">=7.6.x",
-    "npm": ">=4.x"
+  "prettier": {
+    "printWidth": 100,
+    "semi": false,
+    "singleQuote": true,
+    "trailingComma": "es5"
+  },
+  "lint-staged": {
+    "subTaskConcurrency": 1,
```
Comment (skellock) [inline package.json]: You can remove this... if you got it from me, i was struggling one day with `tslint` not formatting "all the file" only "half"... i needed to save several times... i still don't know why that's happening.

**Comment by jamonholmgren** [comment]
@skellock Is it possible to put that ignore rule in the `package.json`? I'm doing some searching but not finding anything.

**Comment by skellock** [comment]
As I understand it, no.  `.prettierignore` settings aren't `prettier` config settings... yet.  i bet there's an issue with this.

**Comment by skellock** [comment]
Also, I think i'm able to put one in my `$HOME` dir so you can say no.

**Comment by skellock** [comment]
I think projects without `prettier` should be forced to put badges on their `readme.md` saying `NO PRETTIER`.  Not vice versa.

**Pull request merged**